x-pack/plugins/monitoring/public/components/metricbeat_migration/flyout/flyout.js

```diff
@@ -0,0 +1,341 @@
+import { STORAGE_KEY, ELASTICSEARCH_CUSTOM_ID } from '../../../../common/constants';
+import { ensureMinimumTime } from '../../../lib/ensure_minimum_time';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  INSTRUCTION_STEP_SET_MONITORING_URL,
+  INSTRUCTION_STEP_ENABLE_METRICBEAT,
+  INSTRUCTION_STEP_DISABLE_INTERNAL
+} from '../constants';
+import { KIBANA_SYSTEM_ID } from '../../../../../xpack_main/common/constants';
+import { Monospace } from '../instruction_steps/components/monospace';
+import { ELASTIC_WEBSITE_URL, DOC_LINK_VERSION } from 'ui/documentation_links';
+
+const storage = new Storage(window.localStorage);
+const ES_MONITORING_URL_KEY = `${STORAGE_KEY}.mb_migration.esMonitoringUrl`;
+const AUTO_CHECK_INTERVAL_IN_MS = 5000;
+const DEFAULT_ES_MONITORING_URL = 'http://localhost:9200';
+
+export class Flyout extends Component {
+  constructor(props) {
+    super(props);
+
+    let esMonitoringUrl = storage.get(ES_MONITORING_URL_KEY);
+    if (!esMonitoringUrl) {
+      esMonitoringUrl = props.monitoringHosts ? props.monitoringHosts[0] : DEFAULT_ES_MONITORING_URL;
+    }
+
+    this.checkInterval = null;
+
+    let activeStep = INSTRUCTION_STEP_SET_MONITORING_URL;
+    if (props.product && props.product.isPartiallyMigrated) {
+      activeStep = INSTRUCTION_STEP_DISABLE_INTERNAL;
+    }
+
+    this.state = {
+      activeStep,
+      esMonitoringUrl,
+      checkedStatusByStep: {
+        [INSTRUCTION_STEP_ENABLE_METRICBEAT]: false,
+        [INSTRUCTION_STEP_DISABLE_INTERNAL]: false,
+      },
+      checkingMigrationStatus: false,
+    };
+  }
+
+  componentWillUpdate(_nextProps, nextState) {
+    // We attempt to provide a better UX for the user by automatically rechecking
+    // the status of their current step, once they have initiated a check manually.
+    // The logic here aims to remove the recheck one they have moved on from the
+    // step
+
+    const thisActiveStep = this.state.activeStep;
+    const nextActiveStep = nextState.activeStep;
+    const nextEnableMbStatus = nextState.checkedStatusByStep[INSTRUCTION_STEP_ENABLE_METRICBEAT];
+    const nowEnableMbStatus = this.state.checkedStatusByStep[INSTRUCTION_STEP_ENABLE_METRICBEAT];
+    const nextDisableInternalStatus = nextState.checkedStatusByStep[INSTRUCTION_STEP_DISABLE_INTERNAL];
+    const nowDisableInternalStatus = this.state.checkedStatusByStep[INSTRUCTION_STEP_DISABLE_INTERNAL];
+
+    const setupInterval = (nextEnableMbStatus && !nowEnableMbStatus) || (nextDisableInternalStatus && !nowDisableInternalStatus);
+    if (setupInterval) {
+      this.checkInterval = setInterval(async () => {
+        await this.checkForMigrationStatus();
+      }, AUTO_CHECK_INTERVAL_IN_MS);
+    }
+
+    const removeInterval = thisActiveStep !== nextActiveStep;
+    if (removeInterval && this.checkInterval) {
+      clearInterval(this.checkInterval);
+      this.checkInterval = null;
+    }
+  }
```

Comment:
Since, you're not clearing the previous timer I think this is creating a new one on every `componentWillUpdate`.

Maybe line 86-97 could be something like this:

```javascript
const setupInterval = (nextEnableMbStatus && !nowEnableMbStatus) || (nextDisableInternalStatus && !nowDisableInternalStatus);
const removeInterval = thisActiveStep !== nextActiveStep;
if (setupInterval || removeInterval) {
  clearInterval(this.checkInterval); //clearInterval can take null or undefined
}

if (setupInterval) {
  this.checkInterval = setInterval(async () => {
    await this.checkForMigrationStatus();
  }, AUTO_CHECK_INTERVAL_IN_MS);
}

```

Also, shouldn't we stop the timer when the component is removed from DOM? eg:
```javascript
componentWillUnmount() {
   clearInterval(this.checkInterval);
}
```

Reply:
++